src/Form/Form.js

```diff
@@ -31,40 +33,52 @@ class Form {
-        this.shouldPromptToStoreCredentials = true
+        this.shouldPromptToStoreData = true
 
         this.submitHandler = () => {
             if (this.handlerExecuted) return
 
-            const credentials = this.getValues()
-            if (credentials.password) {
+            const values = this.getValues()
+            if (this.hasValues(values)) {
                 // ask to store credentials and/or fireproof
-                if (this.shouldPromptToStoreCredentials) {
-                    this.device.storeCredentials(credentials)
+                if (this.shouldPromptToStoreData) {
+                    this.device.storeFormData(values)
                 }
                 this.handlerExecuted = true
             }
         }
 
+        /**
+         * Get values from the form and format them for our device storage
+         * @return {DataStorageObject}
+         */
         this.getValues = () => {
-            const credentials = [...this.inputs.credentials, ...this.inputs.identities].reduce((output, input) => {
-                const subtype = getInputSubtype(input)
-                if (['username', 'password', 'emailAddress'].includes(subtype)) {
-                    output[subtype] = input.value || output[subtype]
-                }
-                return output
-            }, {username: '', password: ''})
-            // If we don't have a username, let's try and save the email if available.
-            if (credentials.emailAddress && !credentials.username) {
-                credentials.username = credentials.emailAddress
-            }
-            delete credentials.emailAddress
-            return credentials
+            const formValues = [...this.inputs.credentials, ...this.inputs.identities, ...this.inputs.creditCards]
+                .reduce((output, inputEl) => {
+                    const mainType = getInputMainType(inputEl)
+                    const subtype = getInputSubtype(inputEl)
+                    let value = inputEl.value || output[mainType]?.[subtype]
+                    if (subtype === 'addressCountryCode') {
+                        value = inferCountryCodeFromElement(inputEl)
+                    }
+                    if (value) {
+                        output[mainType][subtype] = value
+                    }
+                    return output
+                }, {credentials: {}, creditCards: {}, identities: {}})
+
+            return prepareFormValuesForStorage(formValues)
         }
 
-        this.hasValues = () => {
-            const {password} = this.getValues()
-            return !!password
+        /**
+         * Determine if the form has values we want to store in the device
+         * @param {DataStorageObject} [values]
+         * @return {boolean}
+         */
+        this.hasValues = (values) => {
+            const {credentials, creditCards, identities} = values || this.getValues()
+
+            return Boolean(credentials || creditCards || identities)
```

Comment:
These are set to `null` if they don't have enough data to form a usable data entry.

Reply:
yeah - there's a note about increasing type-safety on this stuff below :)